src/main.rs

```diff
@@ -385,8 +385,20 @@ fn configure_app(app: &mut web::ServiceConfig, conf: &MiniserveConfig) {
             .prefer_utf8(true)
             .redirect_to_slash_directory()
             .path_filter(move |path, _| {
-                // deny symlinks if conf.no_symlinks
-                !(no_symlinks && base_path.join(path).is_symlink())
+                if !no_symlinks {
+                    return true;
+                }
+
+                let mut full_path = base_path.clone();
+
+                for component in path.components() {
+                    full_path.push(component);
+                    if full_path.is_symlink() {
+                        return false;
+                    }
+                }
+
+                true
```

Comment:
This logic is now complex enough that I'd like a few comments sprinkled in here.